internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -248,55 +254,30 @@ func (r *NumaflowControllerReconciler) reconcile(
 		return ctrl.Result{}, nil
 	}
 
-	// add Finalizer so we can ensure that we take appropriate action when CRD is deleted
+	// add Finalizer so we can ensure that we take appropriate action when CR is deleted
 	if !controllerutil.ContainsFinalizer(controller, common.FinalizerName) {
 		controllerutil.AddFinalizer(controller, common.FinalizerName)
 	}
 
-	existingDeployment, deploymentExists, err := r.getNumaflowControllerDeployment(ctx, controller)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+	// TODO: we need to be in "Pending" if we haven't deployed yet
 
-	newVersion := controller.Spec.Version
-	newVersionTargetObjs, err := determineTargetObjects(controller, newVersion)
+	_, deploymentExists, err := r.getNumaflowControllerDeployment(ctx, controller)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to determine the target objects for the new version %s: %w", newVersion, err)
-	}
-
-	numaLogger.Debugf("found %d target objects associated with NumaflowController version %s", len(newVersionTargetObjs), newVersion)
-
-	// If a Deployment already exists for the NumaflowController and the new version differs from the current Deployment version,
-	// delete all the NumaflowController child resources and recreate them at the next reconciliation
-	if deploymentExists {
-		currentVersion, err := getControllerDeploymentVersion(existingDeployment)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("error while getting the version of the existing NumaflowController Deployment: %w", err)
-		}
-
-		childResourcesNeedToBeDeleted := controller.Spec.Version != currentVersion
-		if childResourcesNeedToBeDeleted {
-			numaLogger.Debugf("current Deployment image tag=%q differs from desired %q", currentVersion, controller.Spec.Version)
-
-			err := r.deleteNumaflowControllerChildren(ctx, controller, currentVersion, namespace, newVersionTargetObjs)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error deleting NumaflowController child resources: %w", err)
-			}
-
-			// No need to explicitly requeue since a new reconciliation will occur due to changes in watched child resources
-			return ctrl.Result{}, nil
-		}
+		return ctrl.Result{}, err
```

Comment:
previous code looked specifically for "version change" to determine if we need to delete/recreate - now do it any time the manifest has a difference, regardless of whether it's due to "version change"